receiver/zookeeperreceiver/scraper_test.go

```diff
@@ -195,7 +195,7 @@ func TestZookeeperMetricsScraperScrape(t *testing.T) {
 			if !tt.mockZKConnectionErr {
 				go mockZKServer(t, localAddr, tt.mockedZKOutputSourceFilename)
 			}
-			time.Sleep(100 * time.Millisecond)
+			time.Sleep(1 * time.Second)
```

Comment:
Is there any way to avoid a fixed timeout and wait for the right event to happen?

Reply:
The initial update I made to the timeout was based on the hunch that assertions are attempted before the mocked listener is set up. If that's indeed what's causing the flakiness, I believe we can just wait till the listener is setup. PTAL.